### DISCUSSION TITLE
FVDiffusion between meshes

### DISCUSSION THREAD

**Discussion by j-bowhay** [idaholab/moose#20947] [Q&A Modules: General] [answered]
Hi,

I am trying to model a concentric tube heat exchange in moose (without the inputs at right angles as the diagram shows)
<img src="https://user-images.githubusercontent.com/60778417/166644934-24eedbc1-4646-43c4-a2d2-3e40dc13e64d.jpeg" alt="drawing" width="500"/>

I am using INSFV for the fluid flow and FVDiffusion for the conduction. The problem is defined in cylindrical polar about the line of symmetry.

The fluid flow is working as expected with boundary layers developing.
<img width="1303" alt="Screenshot 2022-05-04 at 09 26 33" src="https://user-images.githubusercontent.com/60778417/166646430-ae8cef3c-e86c-415f-a12b-9ce23c1e8af0.png">

However I am not getting any conduction between the different blocks:
<img width="1303" alt="Screenshot 2022-05-04 at 09 31 21" src="https://user-images.githubusercontent.com/60778417/166647131-f3a5dbaf-7603-489f-a6ed-31371ad2f5cd.png">

I would be grateful if anyone has any suggestion to what I am doing wrong.

Many Thanks,
Jake

<details>
<summary>Input File</summary>
<br>

```
# 2d siumulation of water through two concentric pipes

mu=1e-3 # Nsm^-2
rho=997.0 # Kgm^-3
cp_water=4182
k_water=0.598
cp_copper=385
k_copper=386
Re_inner=1000.0
Re_outer=1000.0
pipe_length=1 # m
inner_radius=0.05 # the radius of the inner section m
middle_thickness=0.005 # the thickness of the conductive interface
outer_thickness=0.05 # the thickness of the annulus m
T_inner = 300 # K
T_outer = 290 # K
u_inner_inlet=${fparse (mu * Re_inner)/(2 * inner_radius * rho)}
u_outer_inlet=${fparse (mu * Re_outer)/(2 * inner_radius * rho)}

[GlobalParams]
  rhie_chow_user_object = 'rc'
[]

[UserObjects]
  [rc]
    type = INSFVRhieChowInterpolator
    u = u
    v = v
    pressure = pressure
    block = '0'
  []
[]

[Mesh]
  [inner_mesh]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    xmax = ${pipe_length}
    ymin = 0
    ymax = ${fparse inner_radius}
    nx = 200
    ny = 15
    boundary_name_prefix = 'inner'
  []
  [outer_mesh]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    xmax = ${pipe_length}
    ymin = ${fparse inner_radius + middle_thickness}
    ymax = ${fparse inner_radius + outer_thickness + middle_thickness}
    nx = 200
    ny = 15
    boundary_name_prefix = 'outer'
    boundary_id_offset = 4
  []
  [middle_mesh]
    type = GeneratedMeshGenerator
    dim = 2
    xmin = 0
    xmax = ${pipe_length}
    ymin = ${inner_radius}
    ymax = ${fparse inner_radius + middle_thickness}
    nx = 200
    ny = 2
    boundary_name_prefix = 'middle'
    boundary_id_offset = 8
  []
  [id_middle]
    type = ParsedSubdomainMeshGenerator
    input = middle_mesh
    combinatorial_geometry = 'y > ${inner_radius} & y < ${fparse inner_radius + middle_thickness}'
    block_id = 1
  []
  [combined_mesh]
    type = CombinerGenerator
    inputs = 'inner_mesh outer_mesh id_middle'
  []
  [inner_boundary]
    type = ParsedGenerateSideset
    input = combined_mesh
    combinatorial_geometry = 'abs(y - ${inner_radius}) < 0.0000001'
    replace = true
    new_sideset_name = 'inner_interface'
  []
[]

[Problem]
  fv_bcs_integrity_check = true
  coord_type = 'RZ'
  rz_coord_axis = x
[]

[Variables]
  [u]
    type = INSFVVelocityVariable
    initial_condition = ${u_inner_inlet} #ms^1
    block = '0'
  []
  [v]
    type = INSFVVelocityVariable
    initial_condition = 1e-15 #ms^-1
    block = '0'
  []
  [pressure]
    type = INSFVPressureVariable
    block = '0'
  []
  [T]
    type = INSFVEnergyVariable
    block = '0 1'
  []
[]

[FVKernels]
  [mass]
    type = INSFVMassAdvection
    variable = pressure
    rho = ${rho}
    block = '0'
  []

  [u_advection]
    type = INSFVMomentumAdvection
    variable = u
    rho = ${rho}
    momentum_component = 'x'
    block = '0'
  []
  [u_viscosity]
    type = INSFVMomentumDiffusion
    variable = u
    mu = ${mu}
    momentum_component = 'x'
    block = '0'
  []
  [u_pressure]
    type = INSFVMomentumPressure
    variable = u
    momentum_component = 'x'
    pressure = pressure
    block = '0'
  []

  [v_advection]
    type = INSFVMomentumAdvection
    variable = v
    rho = ${rho}
    momentum_component = 'y'
    block = '0'
  []
  [v_viscosity]
    type = INSFVMomentumDiffusion
    variable = v
    mu = ${mu}
    momentum_component = 'y'
    block = '0'
  []
  [v_pressure]
    type = INSFVMomentumPressure
    variable = v
    momentum_component = 'y'
    pressure = pressure
    block = '0'
  []
  [temp_advection]
    type = INSFVEnergyAdvection
    variable = T
    block = '0'
  []
  [temp_conduction_fluid]
    type = FVDiffusion
    coeff = 'k'
    variable = T
    block = '0 1'
  []
[]

[FVBCs]
  [outer-inlet-u]
    type = INSFVInletVelocityBC
    boundary = '7'
    variable = u
    function = ${u_outer_inlet}
  []
  [outer-inlet-v]
    type = INSFVInletVelocityBC
    boundary = '7'
    variable = v
    function = 0
  []
  [outer-top-no-slip-wall-u]
    type = INSFVNoSlipWallBC
    boundary = '6'
    variable = u
    function = 0
  []
  [outer-top-no-slip-wall-v]
    type = INSFVNoSlipWallBC
    boundary = '6'
    variable = v
    function = 0
  []
  [outer-outlet-p]
    type = INSFVOutletPressureBC
    boundary = '5'
    variable = pressure
    function = 1e-15
  []
  [no-slip-outer-middle-u]
    type = INSFVNoSlipWallBC
    boundary = '4'
    variable = u
    function = 0
  []
  [no-slip-outer-middle-v]
    type = INSFVNoSlipWallBC
    boundary = '4'
    variable = v
    function = 0
  [] 
  [no-slip-inner-middle-u]
    type = INSFVNoSlipWallBC
    boundary = 'inner_top'
    variable = u
    function = 0
  []
  [no-slip-inner-middle-v]
    type = INSFVNoSlipWallBC
    boundary = 'inner_top'
    variable = v
    function = 0
  [] 
  [inner-inlet-u]
    type = INSFVInletVelocityBC
    boundary = 'inner_left'
    variable = u
    function = ${u_inner_inlet}
  []
  [inner-inlet-v]
    type = INSFVInletVelocityBC
    boundary = 'inner_left'
    variable = v
    function = 0
  []
  [inner-outlet-p]
    type = INSFVOutletPressureBC
    boundary = 'inner_right'
    variable = pressure
    function = 1e-15
  []
  [axis-u]
    type = INSFVSymmetryVelocityBC
    boundary = 'inner_bottom'
    variable = u
    u = u
    v = v
    mu = ${mu}
    momentum_component = x
  []
  [axis-v]
    type = INSFVSymmetryVelocityBC
    boundary = 'inner_bottom'
    variable = v
    u = u
    v = v
    mu = ${mu}
    momentum_component = y
  []
  [axis-p]
    type = INSFVSymmetryPressureBC
    boundary = 'inner_bottom'
    variable = pressure
  []
  [T_in_inner]
    type = FVDirichletBC
    variable = T
    boundary = 'inner_left'
    value = ${T_inner}
  []
  [T_in_outer]
    type = FVDirichletBC
    variable = T
    boundary = '7'
    value = ${T_outer}
  []
[]

[Materials]
  [water]
    type = ADGenericFunctorMaterial
    prop_names = 'cp k'
    prop_values = '${cp_water} ${k_water}'
    block='0'
  []
  [copper]
    type = ADGenericFunctorMaterial
    prop_names = 'cp k rho'
    prop_values = '${cp_copper} ${k_copper} 4000'
    block='1'
  []
  [ins_fv]
    type = INSFVEnthalpyMaterial
    temperature = 'T'
    rho = ${rho}
    block='0'
  []
[]

[Executioner]
  type = Steady
  solve_type = 'NEWTON'
  petsc_options_iname = '-pc_type -ksp_gmres_restart -sub_pc_type -sub_pc_factor_shift_type'
  petsc_options_value = 'asm      100                lu           NONZERO'
  nl_max_its = 50       
  l_max_its = 50
  nl_rel_tol = 1e-12
[]

[Outputs]
  exodus = true
[]
```
</details>

**Comment by GiudGiud** [comment]
Hello

So you are trying to model conjugate heat transfer between the solid and the fluid on both sides of the solid?
Or is there a solid surface that connects the two concentric pipes?

Guillaume

**Reply by j-bowhay** [reply]
Thank you for your reply

I am trying to determine the temperature field of the two fluids and the solid given two fluids of different temperatures entering the concentric pipe as shown in the diagram below.
<img width="588" alt="Screenshot 2022-05-05 at 09 34 07" src="https://user-images.githubusercontent.com/60778417/166888291-42545dfe-7fc8-4c2a-b968-fee7370b367d.png">

I have seen the conjugate heat transfer interface kernel however I was under the impression this was for when you want to model heat transfer between a solid and fluid without modelling the fluid flow (as the heat transfer coefficient h encapsulates the boundary layer behaviour.) I would like to include the fluid flow in this model.

Jake

**Reply by GiudGiud** [reply]
Hello

Indeed this interface kernel specifies h, the heat transfer coefficient between the solid and the fluid. It doesnt say you cant model fluid. It's just fixing (well it can still vary, it's a material property) the heat transfer coefficient between the two adjacent cells at the interface. 

But you are right that this encapsulates the boundary layer behavior. We have not implemented any boundary layer treatment save for this interface kernel, which is very simplified treatment for heat transfer. 

Guillaume

**Reply by j-bowhay** [reply]
Is it possible to do this without using the conjugate heat transfer kernel?

Jake

**Reply by GiudGiud** [reply]
Currently not without new code. However MOOSE is designed to be easy to contribute to. What you need to add would fit under an interface kernel, and there's nothing else you need to worry about to get compatibility with everything else in the module

**Reply by lindsayad** [reply]
What happens if you refine this mesh? Having T = 0 in the middle block with FVDiffusion active on the whole domain doesn't make any sense. It feels like something wonky is going on. I should run this input

**Comment by lindsayad** [comment, marked as answer]
@j-bowhay the reason you are seeing no conduction is because you are using a `CombinerGenerator`. This class makes no attempts to stitch the meshes together, so the elements on the different mesh blocks do not form neighbor connections with the other mesh blocks. You could use a `CartesianMeshGenerator` or one `GeneratedMeshGenerator` and then a couple `SubdomainBoundingBoxGenerators` etc. Even a couple `StitchedMeshGenerators` but I wouldn't recommend that for this simple of a geometry.

Because elements don't form neighbor connections with the neighboring mesh blocks, the FV system thinks the mesh interfaces are actually external boundaries with neither flux nor Dirichlet boundaries specified and consequently treats those boundaries as natural boundaries (e.g. no flux)
